client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -72,6 +83,15 @@ const hosting: Flow = {
 			}
 
 			switch ( _currentStepSlug ) {
+				case 'domains': {
+					if ( plan && isDotComPlan( { product_slug: plan } ) ) {
+						setPlanCartItem( {
+							product_slug: plan,
+						} );
+						return navigate( 'createSite' );
+					}
```

Comment:
I think this would benefit from a comment like `When plan is already supplied as a query parameter, add it to cart, and skip plans step`.